### PR TITLE
Use dscp-node types package

### PR DESCRIPTION
**Pull request by jonmattgray**
Use the `@digicatapult/dscp-node` package to get types. Always uses latest version of the package (and latest `dscp-node` image for testing)

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+registry=https://registry.npmjs.org/
+//npm.pkg.github.com/:_authToken=${GITHUB_PACKAGE_TOKEN}
```
Comment (dblane-digicatapult) [inline .npmrc:2]: Is this needed?  Its a public repository, so we shouldn't need to include a token to get the package.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -121,6 +121,8 @@ jobs:
             ${{ runner.OS }}-
       - name: Install Packages
         run: npm ci
+        env:
+          GITHUB_PACKAGE_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (dblane-digicatapult) [inline .github/workflows/release.yml:125]: Is this actually required?

File: Dockerfile [hunk 3]
```diff
@@ -17,7 +17,7 @@ WORKDIR /dscp-identity-service
 RUN npm -g install npm@8.x.x
 
 COPY . .
-RUN npm ci --production
+RUN --mount=type=secret,id=github GITHUB_PACKAGE_TOKEN=$(cat /run/secrets/github) npm ci --production
```
Comment (dblane-digicatapult) [inline Dockerfile:20]: I'm not sure this is needed

**Review by dblane-digicatapult — COMMENTED** [review]
I'm not at all certain that we need to auth to the github npm registry in order to get a public package.  Can you please verify if this is actually required?

File: .github/workflows/release.yml (hunk 2 above)
Comment (jonmattgray) [reply .github/workflows/release.yml:125]: seems not!

File: .npmrc (hunk 1 above)
Comment (jonmattgray) [reply .npmrc:2]: I think this one is, `npm i` doesn't work locally without, even for 'public' packages
https://github.community/t/download-from-github-package-registry-without-authentication/14407

File: Dockerfile (hunk 3 above)
Comment (jonmattgray) [reply Dockerfile]: looks like it is, `docker build` doesn't work locally for me without it either
https://github.com/digicatapult/dscp-identity-service/runs/6021152096?check_suite_focus=true

File: .github/workflows/release.yml (hunk 2 above)
Comment (jonmattgray) [reply .github/workflows/release.yml:125]: actually I removed the cache and it fails without env

**Review by dblane-digicatapult — APPROVED** [review]

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**